edx_solutions_api_integration/users/serializers.py

```diff
@@ -57,27 +55,37 @@ def get_courses_enrolled(self, user):
 
     def get_user_roles(self, user):
         """ returns list of user roles """
-        queryset = CourseAccessRole.objects.filter(user=user)
+        roles = user.courseaccessrole_set.all()
         if 'course_id' in self.context:
             course_id = self.context['course_id']
-            queryset = queryset.filter(course_id=course_id)
+            roles = [role.role for role in roles if role.course_id == course_id]
```

Comment:
How about renaming roles as 'course_access_roles'? In the list comprehension we can have
`[course_access_role.role for course_access_role in course_access_roles if course_access_role.course_id == course_id]`.
This can improve readability in code.